src/test/java/com/epam/automation/training/TestListener.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.automation.training;
+
+import com.epam.automation.training.annotations.CustomAnnotations.APITests;
+import com.epam.automation.training.annotations.CustomAnnotations.UITests;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+public class TestListener implements ITestListener {
+
+    Method[] tests;
+
+    public void onTestStart(ITestResult result) {
+        tests = result.getTestClass().getRealClass().getMethods();
+    }
+
+    public void onTestSuccess(ITestResult result) {
+    }
+
+    public void onTestFailure(ITestResult result) {
+    }
+
+    public void onTestSkipped(ITestResult result) {
+    }
+
+    public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
+    }
+
+    public void onTestFailedWithTimeout(ITestResult result) {
+        this.onTestFailure(result);
+    }
+
+    public void onStart(ITestContext context) {
+    }
+
+    public void onFinish(ITestContext context) {
+        getUiTestsInfo();
+    }
+
+    private void getUiTestsInfo() {
```

Comment:
I'm not sure it works correctly in all cases because you update tests in onTestStart(). Let's discuss